resources/assets/js/components/teams/TeamMembershipButton.vue

```diff
@@ -0,0 +1,89 @@
+<template>
+    <div>
+        <button
+                type="button" class="btn btn-secondary"
+                :disabled="!this.user.is_active || !this.team.self_serviceable"
+                v-if="memberOfTeam" v-on:click="changeMembership('leave')">
+            Leave
+        </button>
+        <button
+                type="button" class="btn btn-primary"
+                :disabled="!this.user.is_active || !this.team.self_serviceable"
+                v-else v-on:click="changeMembership('join')">
+            Join
+        </button>
+        <em v-if="!this.user.is_active">&nbsp;&nbsp;<small>You must be an active member to join/leave teams.</small></em>
+        <em v-if="!this.team.self_serviceable">&nbsp;&nbsp;<small>Join/leave is restricted for this team.</small></em>
```

Comment:
Rephrase this to be more helpful - "Contact an administrator to leave this team".

If it's not too difficult, either say "join" or "leave" in the text. The slash is not necessary since the application knows which makes sense for this particular team.

Reply:
I tried your proposed phrasing or something close to it and it was too long for the space - caused the footer panel to be larger than necessary. I'll fiddle with it some more.

I'll update the text accordingly, shouldn't be too terrible.